django-stubs/contrib/admin/options.pyi

```diff
@@ -129,6 +129,7 @@ class BaseModelAdmin(Generic[_ModelT]):
     def get_prepopulated_fields(self, request: HttpRequest, obj: _ModelT | None = ...) -> dict[str, Sequence[str]]: ...
     def get_queryset(self, request: HttpRequest) -> QuerySet[_ModelT]: ...
     def get_sortable_by(self, request: HttpRequest) -> _DisplayT[_ModelT]: ...
+    @deprecated("The None value for the request parameter will be removed in Django 6.0.")
```

Comment:
This one should be removed. `@deprecated` can only be used if the whole function is deprecated, not changes to types for individual parameters. This will cause the deprecation to be shown even if `request` is not `None`.

Reply:
Alternatively, [deprecated overloads](https://typing.readthedocs.io/en/latest/spec/directives.html#example) can be used.

Reply:
Of course, yes!